### PR TITLE
feat: add debug prep scripts and configuration for server

### PR DESCRIPTION
**Pull request by tommartensen**
See README for usage.

Please test it before approving!
You can use `gcloud container clusters get-credentials infra-pr-1104 --zone us-central1-a --project acs-team-temp-dev` to point to a cluster with workflow controller and secrets prepared.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,26 @@ toolchain, it is also possible to rely on CI. CI will lint, build and push the
...
+
+To debug a CLI command, update the VS Code launch configuration "Debug CLI" with the desired command and update the `INFRA_TOKEN` value if necessary.
+You may also point to a different infra server with the `--endpoint` flag.
+
+#### Server
+
+To debug the server, you need to fulfil the prerequisites first.
+
+1. Have an authenticated `gcloud` CLI.
+1. Have your `KUBECONFIG` point to a cluster where Argo Workflows and the ConfigMaps and Secrets for infra are deployed. This is most easily achieved by connecting to a PR cluster or deploying infra with `ENVIRONMENT=<DEVELOPMENT,PRODUCTION> make install-argo clean-argo-config helm-deploy` to a new or local cluster. This cluster will only be used to run workflows.
```
Comment (gavin-stackrox) [inline README.md:35]: What about the infra-server deployment? I'm guessing this needs to be deleted so as not to conflict with the local dev server.

**Review by gavin-stackrox — APPROVED** [review]
I'm looking forward to trying this out for my next PR. If someone else gets to try it out before merge that is great but otherwise I'd say :shipit:

File: README.md (hunk 1 above)
Comment (tommartensen) [reply README.md:35]: No it doesn't have to be deleted. infra-server is stateless and all data is in Argo workflow K8s objects. 
The point is that you can use the local debugging server against any existing infra cluster, ie also against dev & prod if you want to debug a bug 🐛

File: scripts/local-dev/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+echo "INFO: Download secrets"
+ENVIRONMENT=development make secrets-download
+
+echo "INFO: Place configuration with rendered values"
+go run scripts/local-dev/main.go
+
+echo "INFO: Replace /configuration/ -> ../../configuration/ in infra.yaml"
+sed -i 's/\/configuration\//..\/..\/configuration\/tls\//g' configuration/infra.yaml
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply scripts/local-dev/prepare.sh:17]: > Do we expect gnu sed?

yes, good catch. I updated the script to fail if `sed` is not GNU and added it as a prerequisite in the README. 
IIRC it is a requirement for some stackrox scripts already, so most people should have it.

**Pull request merged**